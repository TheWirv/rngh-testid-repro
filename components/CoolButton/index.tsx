import {View, Text, StyleSheet} from 'react-native';
import {TouchableOpacity} from 'react-native-gesture-handler';
import type {FunctionComponent} from 'react';
import type {StyleProp, ViewStyle} from 'react-native';

const CoolButton: FunctionComponent<{
  amount: number;
  onPress: () => void;
  theme: 'dark' | 'light';
  style?: StyleProp<ViewStyle>;
  containerStyle?: StyleProp<ViewStyle>;
}> = ({children, amount, onPress, theme, style, containerStyle}) => {
  return (
    <TouchableOpacity
      testID="increaseAmount"
      style={[styles[`button-${theme}`], style]}
      containerStyle={containerStyle}
      onPress={onPress}>
      <View style={styles.container}>
        <View style={{flex: 1}}>{children}</View>
        <Text style={styles.amount}>{amount}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    paddingRight: 16,
  },
  amount: {
    fontSize: 16,
    fontWeight: 'bold',
  },
  'button-light': {
    backgroundColor: '#5555EE',
  },
  'button-dark': {
    backgroundColor: '#CC2222',
  },
});

export default CoolButton;
