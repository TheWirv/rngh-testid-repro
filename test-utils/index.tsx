import {SafeAreaProvider} from 'react-native-safe-area-context';
import {render as testingLibRender} from '@testing-library/react-native';
import type {RenderOptions} from '@testing-library/react-native';
import type {ReactElement} from 'react';

const AllTheProviders = ({children}: {children: ReactElement}) => (
  <SafeAreaProvider>{children}</SafeAreaProvider>
);

// re-export everything
export * from '@testing-library/react-native';

// override render method
export function render(ui: ReactElement, options?: RenderOptions) {
  return testingLibRender(ui, {wrapper: AllTheProviders, ...options});
}
