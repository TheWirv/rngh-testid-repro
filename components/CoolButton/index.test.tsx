import type {FunctionComponent} from 'react';
import {render, fireEvent} from '../../test-utils';
import CoolButton from './index';

describe('Component test: CoolButton', () => {
  let amount = 1;
  const onPress = () => {
    amount++;
  };

  const TestCoolButton: FunctionComponent = () => (
    <CoolButton theme="dark" amount={amount} onPress={onPress} />
  );

  test('Correct rendering', () => {
    const tree = render(<TestCoolButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Functionality: `increaseAmount`', () => {
    const {getByTestId} = render(<TestCoolButton />);
    fireEvent.press(getByTestId('increaseAmount'));
    fireEvent.press(getByTestId('increaseAmount'));

    expect(amount).toEqual(3);
  });
});
