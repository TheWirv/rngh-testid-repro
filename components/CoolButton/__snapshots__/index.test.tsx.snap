// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component test: CoolButton Correct rendering 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  exclusive={true}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  testID="increaseAmount"
>
  <View
    accessible={true}
    collapsable={false}
    style={
      Object {
        "backgroundColor": "#CC2222",
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "paddingRight": 16,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        1
      </Text>
    </View>
  </View>
</RNGestureHandlerButton>
`;
